projects/hipblaslt/CMakeLists.txt

```
@@ -563,20 +556,23 @@ if(HIPBLASLT_ENABLE_COVERAGE)
       find_program(
         LLVM_PROFDATA
         llvm-profdata
-        HINTS llvm/bin
         REQUIRED
+        HINTS ${ROCM_PATH}/llvm/bin
+        PATHS /opt/rocm/llvm/bin
```

```suggestion
        PATH_SUFFIXES llvm/bin
```
We should avoid hardcoding absolute paths like /opt/rocm or relying on custom variables like ROCM_PATH if possible. Since we already rely on CMAKE_PREFIX_PATH to find other dependencies (like HIP), we can use PATH_SUFFIXES to find the LLVM tools within those same prefixes. For example, TheRock uses a non-standard install location and explicitly unsets ROCM_PATH, so with PATH_SUFFIXES we can support coverage without the dependency on a (possibly) undefined ROCM_PATH and allow relocatable toolchains.

---

So we use this to find the `llvm-cov` and `llvm-profdata` executables that match the version of our clang.  In CI, we only use amdclang, which is llvm version 20, so we need to use the same binaries.

I agree with what you say about hardcoding, but I don't think your suggestion will work with our CI as designed.  A better approach would be to pass in `-DLLVM_COV_BIN_DIR` during build, and then Math CI will pass that in and it will be more flexible for others that want to make a coverage report.

That said, pretty much every project has these values hardcoded, so we will need to address multiple projects.  Is it ok to leave this code as is and I will create a ticket to ensure we revisit it?

---

Okay, we can make a ticket to make sure this gets revisited.